### PR TITLE
fix: missing last day of month

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

- Fix that the last day in Oktober was always shown as unavailable
This happened when availability was in Europe/London timezone

- Fix that the last day of November was always shown as unavailable
This happened also for other timezones

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

I recommend testing it in staging instead of local, as I experienced some different behaviors on local 

- Reproduce the 2 bugs as described before my fix
- If you were able to reproduce it, see if it is fixed with my fixes in this PR (last day of octber and November should be shown as available)
- Check if availability starts and ends at the correct time on days were DST happens

### CODE REVIEW

File: packages/lib/date-ranges.test.ts [hunk 1]
```diff
@@ -29,6 +29,25 @@ describe("processWorkingHours", () => {
...
+    const timeZone = "Europe/London";
+
+    const dateFrom = dayjs().month(9).date(24); // starts before DST change
+    const dateTo = dayjs().startOf("day").month(10).date(1); // first day of November
+
+    const results = processWorkingHours({ item, timeZone, dateFrom, dateTo });
+
+    const lastAvailableSlot = results[results.length - 1];
+
+    expect(lastAvailableSlot.start.date()).toBe(31);
```
Comment (CarinaWolli) [inline packages/lib/date-ranges.test.ts:48]: This failed before, the last day was 30

**Review by zomars — APPROVED** [review]
Nice job @CarinaWolli LGTM

**Comment by zomars** [comment]
### Merge activity

* **Oct 11, 2:30 PM**: @zomars merged this pull request with [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/11820).

**Comment by zomars** [comment]
Hey @CarinaWolli IDK why locally one of these tests are failing 

image

**Pull request merged**